pkg/daemon/update.go

```diff
@@ -1513,6 +1513,16 @@ func createSSHKeyDir(authKeyDir string) error {
 		return exec.Command("runuser", "-u", constants.CoreUserName, "--", "mkdir", "-m", "0700", "-p", dir).Run()
 	}
 
+	// Check the existence of the /home/core/.ssh dir before creating a new one
+	// via runuser core by hand. Delete if any in case the dir is created at a wrong
+	// time and under the wrong user (root), and let the MCD recreate it.
+	// Serve as a workaround for https://issues.redhat.com/browse/OCPBUGS-11832
+	if _, err := os.Stat(constants.CoreUserSSHPath); err == nil {
+		if err := os.RemoveAll(constants.CoreUserSSHPath); err != nil {
+			return err
```

Comment:
maybe format this error if it fails so we know which path failed to remove, otherwise this will look generic.

something like `return fmt.Errorf(err, "could not remove %s, constants.CoreUserSSHPath)`